services/api/src/unified_graphics/diag.py

```diff
@@ -264,3 +264,24 @@ def wind(
         )
         for idx in range(data.dims["nobs"])
     ]
+
+
+def magnitude(dataset: List[Observation]) -> List[Observation]:
+    return [
+        Observation(
+            obs.variable,
+            obs.variable_type,
+            obs.loop,
+            adjusted=vector_magnitude(obs.adjusted.u, obs.adjusted.v)
+            if isinstance(obs.adjusted, Vector)
+            else obs.adjusted,
+            unadjusted=vector_magnitude(obs.unadjusted.u, obs.unadjusted.v)
+            if isinstance(obs.unadjusted, Vector)
+            else obs.unadjusted,
+            observed=vector_magnitude(obs.observed.u, obs.observed.v)
+            if isinstance(obs.observed, Vector)
+            else obs.observed,
+            position=obs.position,
+        )
```

Comment:
The syntax here took me a minute to decipher. This is just an inline if/else statement like below, right? It's kind of funny seeing the inline if/else spread across multiple lines. I suppose we have PEP8/Black to thank on that one. 

```python
Observation(
            obs.variable,
            obs.variable_type,
            obs.loop,
            adjusted=vector_magnitude(obs.adjusted.u, obs.adjusted.v) if isinstance(obs.adjusted, Vector) else obs.adjusted,
            unadjusted=vector_magnitude(obs.unadjusted.u, obs.unadjusted.v) if isinstance(obs.unadjusted, Vector) else obs.unadjusted,
            observed=vector_magnitude(obs.observed.u, obs.observed.v) if isinstance(obs.observed, Vector) else obs.observed,
            position=obs.position,
        )
```

Would do something like this for clarity be okay?
```python
Observation(
            obs.variable,
            obs.variable_type,
            obs.loop,
            adjusted=( 
                vector_magnitude(obs.adjusted.u, obs.adjusted.v)
                if isinstance(obs.adjusted, Vector)
                else obs.adjusted
            ),
            unadjusted=(
                vector_magnitude(obs.unadjusted.u, obs.unadjusted.v)
                if isinstance(obs.unadjusted, Vector)
                else obs.unadjusted
             ),
            observed=(
                vector_magnitude(obs.observed.u, obs.observed.v)
                if isinstance(obs.observed, Vector)
                else obs.observed
             ),
            position=obs.position,
)
```

Reply:
Yeah, Black formatted it that way. The only reason its even in there is because mypy complained. I’ll try wrapping it with the `()`. If Black accepts that, it’s definitely easier to read. Although maybe I should just rewrite that to make it more clear that we’re checking for scalar values…

Reply:
Oh, interesting. I wouldn't have thought mypy would be particularly involved there. Though - I guess it is good that it's making sure you're getting scalars in there and not vectors.

Reply:
Alright, I opted to refactor the code instead of just reformatting it. Let me know if you think it’s still unclear.

Reply:
That's a really neat use of a generator! I keep forgetting they're in Python. Looks good to me!